README.md

```diff
@@ -1,51 +1,161 @@
+
+To build with the bare metal profile, the application configuration file must contain the following:
 ```
-mbed import mbed-os-example-blinky-baremetal
-cd mbed-os-example-blinky-baremetal
+{
+    "requires": ["bare-metal"]
+}
 ```
 
-### Now compile
+Note: For convenience, a few configuration files are present in the `configs` directory so you can try some of the different configurations shown in this example.
+
+### Futher optimisations
+
+#### Linking with smaller C libraries
+
+Both the `ARM` and `GCC_ARM` toolchains support optimised versions of their C standard libraries, microlib and newlib-nano respectively and we recommend using them with the bare metal profile.
+
+To build with the smaller C libraries, the application configuration file needs to be modified as follows:
+```
+{
+    "target_overrides": {
+        "*": {
+            "target.c_lib": "small"
+        }
+    }
+}
+```
+#### Using Mbed minimal printf library
 
-Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler:
+Mbed OS offers a smaller printf() alternative. The [minimal printf](https://github.com/ARMmbed/mbed-os/blob/master/platform/source/minimal-printf/README.md) library implements a subset of the `v/s/f/printf` function family and floating points can be disabled to save additional code size.
 
+To build with the minimal printf library and disable floating points printing, the application configuration file needs to be modified as follows:
 ```
-mbed compile -m K64F -t ARM
+{
+    "target_overrides": {
+        "*": {
+            "target.printf_lib": "minimal-printf",
+            "platform.minimal-printf-enable-floating-point": "false"
+        }
+    }
+}
 ```
 
-Your PC may take a few minutes to compile your code. At the end, you see the following result:
+#### Using a minimal console
+
+If your application only needs basic unbuffered I/O operations then you can save additional memory by using a configuration of the platform library which removes file handling functionality from the [system I/O retarget code](https://github.com/ARMmbed/mbed-os/blob/master/platform/source/mbed_retarget.cpp).
 
+To build with the minimal console functionality, the application configuration file needs to be modified as follows:
 ```
-[snip]
-Image: ./BUILD/K64F/GCC_ARM/mbed-os-example-blinky-baremetal.bin
+{
+    "target_overrides": {
+        "*": {
+            "platform.stdio-minimal-console-only": "true"
+        }
+    }
+}
 ```
 
-### Program your board
+#### Memory comparison
+
+Example configuration files are provided under the `configs/` directory. You may override the `mbed_app.json` with either of these using the `--appconfig` optional argument.
```

Comment:
Is that useful (providing additional config for each option)? Or is it just noise and complication?

Reply:
I can remove indeed. I was in two minds about it.